### DISCUSSION TITLE
Support for Optimistic Concurrency?

### DISCUSSION THREAD

**Discussion by refactorthis** [typedorm/typedorm#259] [Q&A] [answered]
Hi,

Not sure how best to implement a version attribute for optimistic concurrency. Does typedorm support this?
For now I'm incrementing the version number then adding a conditional write to ensure that the version number is one less than the current. Seems to do the trick, but is there a better way?

**Comment by whimzyLive** [comment, marked as answer]
Hi @refactorthis,

Typedorm doesn't have any native features to support optimistic concurrency as it is offloading most of these operational responsibilities to DynamoDB itself.

The way you are currently handling the optimistic concurrency (via conditional write) will also work with TypeDORM. 

You can provide the condition in the [options.where](https://github.com/typedorm/typedorm/blob/main/docs/api.md#entitymanagerupdate) property and TypeDORM will generate the appropriate condition expression for you.
